ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,134 @@ const base = url.substring(0, url.lastIndexOf('/'));
+      expect(layout.element.$$('span')).to.exist;
+      expect(layout.element.$$('span').innerText).to.equal('dummy');
+      expect(layout.element.$$('nuxeo-input')).to.exist;
+    });
+
+    test('Should stamp the layout when model is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      expect(layout.element.$$('span')).to.exist;
+      expect(layout.element.$$('nuxeo-input')).to.exist;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      await flush();
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      sinon.spy(layout.element, 'validate');
+      const nuxeoInput = layout.element.$$('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.true;
+      expect(nuxeoInput.invalid).to.be.false;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+      expect(layout.element.validate.calledOnce).to.be.true;
+    });
+
+    test('Should invalidate the layout when required input is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      const nuxeoInput = layout.element.$$('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.false;
+      expect(nuxeoInput.invalid).to.be.true;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+    });
+
+    test('Should invalidate the layout when custom validation is invalid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout
+          href="${base}/layouts/dummy-layout.html"
+          model='{"text": "invalid", "data": "foo"}'
+        ></nuxeo-layout>
+      `);
+      await flush();
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      sinon.spy(layout.element, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.false;
+      expect(layout.element.validate.calledOnce).to.be.true;
```

Comment:
And we don't call the `validate` method in any of our nodes, correct? (span and nuxeo-input)